python_ta/upload.py

```diff
@@ -0,0 +1,87 @@
+                err_as_dict[key] = []
+                info_set = current_type.get(key)
+                for msg in info_set.messages:  # Iterates over the messages for each error of the given code
+                    info = {k: getattr(msg, k) for k in error_info}
+                    err_as_dict[key].append(info)
+    return err_as_dict
+
+
+def upload_to_server(errors, config: Dict[str, str], default: Dict[str, str], url: str, version: str, paths=None):
+    """Send POST request to server with formatted data."""
+    unique_id = get_hashed_id()  # Hashing just the mac address portion of the uuid
+    files = []
+    if paths:
+        for path in paths:
+            f = open(path, 'rb')
+            files.append(f)
+    upload = {str(i): f for i, f in enumerate(files)}  # requests.post() requires passing a dict
+    # upload is an empty dict in the case that paths is empty
+    errors_dict = errors_to_dict(errors)
+    to_json = {'errors': errors_dict}
+    if config != default:  # checks for (dictionary) equality of settings between config used and default config
+        to_json['cfg'] = config
+    payload = json.dumps(to_json)
+    try:
+        response = requests.post(
+            url=url,
+            files=upload,
+            data={'id': unique_id,
+                  'version': version,
+                  'payload': payload
+                  }
+        )
+        response.raise_for_status()
+    except requests.HTTPError as e:
+        print('[ERROR] Upload failed')
+        if e.response.status_code == 400:
+            print('[ERROR] HTTP Response Status 400: Client-side error, likely due to improper syntax. '
+                  'Please report this to your instructor (and attach the code that caused the error).')
+            raise e
+        elif e.response.status_code == 403:
+            print('[ERROR] HTTP Response Status 403: Authorization is currently required for submission.')
+            raise e
+        elif e.response.status_code == 500:
+            print('[ERROR] HTTP Response Status 500: The server ran into a situation it doesn\'t know how to handle. ')
+            print('Please report this to your instructor (and attach the code that caused the error).' )
+            raise e
+        elif e.response.status_code == 503:
+            print('[ERROR] HTTP Response Status 503: The server is not ready to handle your request. ')
+            print('It may be down for maintenance.')
+            raise e
+        else:
+            print('[ERROR] Error message: "{}"'.format(e))
+            raise e
+    except requests.ConnectionError as e:
+        print('[ERROR] Upload failed')
+        print('[ERROR] Error message: Connection timed out. This may be caused by your firewall, or the server may be '
+              'temporarily down.')
+    for f in files:  # Closing files after uploading
+        f.close()
+
+
+def get_hashed_id():
```

Comment:
Add a return type annotation here. Also, the docstring needs to be updated because you changed the functionality.